packages/libs/api/icy-graphql-client/src/client/icyGraphqlClient.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  ApolloClient,
+  NormalizedCacheObject,
+  QueryOptions,
+} from '@apollo/client';
+import { removeNodesAndEdges } from '../utils/removeNodesAndEdges';
+
+export class IcyGraphQLClient {
+  constructor(public apolloClient: ApolloClient<NormalizedCacheObject>) {}
+
+  /**
+   * @todo improve typing here
+   */
+  async query(options: QueryOptions<any, any>): Promise<any> {
+    const result = await this.apolloClient.query(options);
+
+    if (result?.data && typeof result?.data === 'object') {
+      result.data = removeNodesAndEdges(result.data);
```

Comment:
I think you can do this in an ApolloLink and not have to have this `icyGraphQLClient`. Basically as middleware and apply to all responses

Reply:
Yup, I had the same thought going off of https://www.apollographql.com/docs/react/networking/advanced-http-networking/#modifying-response-data, but it seems like Apollo's in memory cache is run in between links (didn't get to dive fully on this, so might be slightly wrong on relation of errors) doesn't love that the actual structure and not just the scalar properties are modified https://github.com/apollographql/apollo-client/issues/8677, so was getting errors left and right. 
On a side learning, I also intended on directly extending Apollo's client and overriding the query method calling super and then modifying on the last mile, but it was causing typing issues as I learnt that you can't actually override the function signature (i.e. returning flattened data type) from the parent class; that's the reason for the total custom client that in itself delegates to the apollo client

Reply:
>Should we make the standard apollo client private in IcyGraphQLSDK? 

This is something I actually gave some thought, my feeling was that it could be helpful to expose the apollo client we build so that we can in the future instruct users to access that apollo client if they want specific data (and hence save them the hassle of setting up the http/error/auth links if they just want to get bootstrapped fast)
On second thought though, we could make it private as a property and set a custom named getter method if you think that'd be more clear? My logic on naming was a direct consequence of assuming that we'd be exposing both the "apollo" raw client and the "icy" higher level client

Note of color as well though, I was thinking that on our "Icy" higher level client we will probably not be 1:1 matching functionality the same as Apollo client, i.e. one improvement I was thinking along this line (longer term) is that we could return a "IcyResponse" instance as a superset of ApolloResponse with flattened data and magic pagination methods (i.e. response.nextPage())

Reply:
> When do you think users will need to use both clients?

i.e. when a user wants to fetch some specific data or filter that is not fully covered by our abstraction SDK, they can access our raw apollo client and forward the sql query without having to do the auth/http/error link setup themselves

> The icy one should just be a "superset" of the apollo one, no?

Technically no, because of not directly extending the apollo client as a consequence of method signatures being modified, but also it might be an opportunity to bring a more "graphql for dummies" sort of ORM approach 

>Note of color as well though, I was thinking that on our "Icy" higher level client we will probably not be 1:1 matching functionality the same as Apollo client, i.e. one improvement I was thinking along this line (longer term) is that we could return a "IcyResponse" instance as a superset of ApolloResponse with flattened data and magic pagination methods (i.e. response.nextPage())